src/components/elements/quer-image-carousel.tsx

```diff
@@ -0,0 +1,141 @@
+import React, { MouseEventHandler, ReactElement } from 'react'
+import { graphql, useStaticQuery } from 'gatsby'
+import styled, { css } from 'styled-components'
+import { GatsbyImage, StaticImage, getImage } from 'gatsby-plugin-image'
+import type { ImageDataLike, IGatsbyImageData } from 'gatsby-plugin-image'
+
+export type QueryImageProps = {
+    alt: ReactElement | string
+    className?: string
+    data: ImageDataLike | IGatsbyImageData
+    keyName: string
+    height?: string
+    width?: string
+    max_width?: string
+    loading?: 'eager' | 'lazy'
+    disable_transition?: boolean
+    onMouseOver?: MouseEventHandler<HTMLDivElement>
+    onMouseOut?: MouseEventHandler<HTMLDivElement>
+    onClick?: MouseEventHandler<HTMLDivElement>
+}
+
+export type ImageWrapperProps = {
+    bimage: string
+    width: string
+    height: string
+    className?: string
+    loading: 'eager' | 'lazy'
+    disable_transition?: boolean
+}
+
+const query = graphql`
+    query {
+        sprite_hero: file(relativePath: { eq: "home/MergedImages-new.png" }) {
+            ...homePageHeroFadeIn
+        }
+        background_hero: file(relativePath: { eq: "home/hero_bg.png" }) {
+            ...homePageHeroFadeIn
+        }
+    }
+`
+
+export const ImageWrapper = styled.div<ImageWrapperProps>`
+    & .gatsby-image-wrapper {
+        width: ${(props) => props.width || '100%'};
+        height: ${(props) => props.height};
+    }
+    .gatsby-image-wrapper [data-main-image] {
+        ${({ loading, disable_transition }) => {
+            if (disable_transition && loading === 'eager') {
+                return css`
+                    transition: none;
+                    opacity: 1;
+                `
+            }
+        }};
+    }
+    .gatsby-image-wrapper img {
+        height: 700px !important;
+    }
+    .gatsby-image-wrapper .hero1 {
+        background: url(${(props) => props.bimage}) 120px 35px;
+        background-repeat: no-repeat;
+        background-size: 1911px;
+    }
+    .gatsby-image-wrapper .hero2 {
+        background: url(${(props) => props.bimage}) -590px 35px;
+        background-repeat: no-repeat;
+        background-size: 1911px;
+    }
+    .gatsby-image-wrapper .hero3 {
+        background: url(${(props) => props.bimage}) -1280px 35px;
+        background-repeat: no-repeat;
+        background-size: 1911px;
+    }
+`
+
+const QueryImageCarousel = ({
+    alt,
+    className,
+    data,
+    keyName,
+    disable_transition = false,
+    height,
+    loading = 'lazy',
+    onClick,
+    width,
+    ...props
+}: QueryImageProps) => {
+    const image = getImage(data)
+    const heroe_data = useStaticQuery(query)
+    //console.log('heroe_data heroe_data heroe_data1', heroe_data)
```

Comment:
let's not forget this console log

Reply:
Thanks for pointing this out Al Amin, It is a draft PR only. will clean the code before making real PR.